src/System.Net.Sockets/tests/FunctionalTests/SocketAsyncExtensions.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    internal static class SocketAsyncExtensions
```

Comment:
Yes, these are more extension methods: I don't care whether or not they remain extensions methods, but I do plan on keeping the functionality in discrete static methods.